Server improvements + Extra Attribute display

fosskers opened the pull request:
In learning the server code, I've updated it to:
- Yield 404 on a tile outside the current bounds, instead of an ErrorTile (Rob's suggestion)
- Use less verbose naming conventions
- Hold route definitions and their handle code in separate functions for easier readability

Note that even with an ErrorTile/404 returned on a Tile outside the current bounds, Leaflet calling these in the first place feels wasteful. Something to consider for the next sprint.

fosskers committed:
Squashed commit of the following:

commit 8f5650697e4b7b5bd0281fc9c64cb6180ebf1d34
Author: Colin Woodbury <colingw@gmail.com>
Date:   Fri May 13 10:59:28 2016 -0700

    Small lints before experiment

commit b617dd366e779e5217a3efcd159252ecbec30791
Author: Colin Woodbury <colingw@gmail.com>
Date:   Thu May 12 16:10:53 2016 -0700

    Some structural linting

    - Warning: Ensime's `refactor-source` should probably not, in general, be used.
    It causes too many lines to change where there was no difference in
    functionality. This could lead to merge headaches later.
    - I say this because I'm guilty of doing this in this commit.
    Should be avoided in the future.

fosskers committed:
(server) Broke up route handling code into seperate functions

fosskers committed:
(server) Avoid `apply` magic with `KryoRegistration`

fosskers committed:
(server) Get class names in a better way to squash warning

- The class name of Avro's `Schema.Field.Order` was being called
in a hacky way.

fosskers committed:
(server) Remove `EmptyTile`

fosskers committed:
(server) Remove `errorTile` API endpoint

fosskers committed:
(server) Less verbose naming

moradology commented on server/src/main/scala/geotrellis/admin/server/AdminActor.scala:
```diff
@@ -0,0 +1,249 @@
+    }
+  }
+
+  def tms = handleExceptions(missingTileHandler) {
```
This is definitely a good change. Much cleaner

fosskers committed:
(client) Linting + major cleaning of imports

fosskers committed:
(server) New `attributes` endpoint for getting extra non-metadata attributes

- Attributes called from S3 in this way are cached.

fosskers committed:
(server) Fetch metadata directly for `bounds` endpoint

- Previously, this was fetching the entire tile, and extracting its metadata
from there. Now this call is faster, and the results are also cached.

fosskers committed:
(server) Reinstate `errorTile` endpoint

- It turns out the the Leaflet needed it for filling in dead tiles.
The logs are no longer overun by 404s.

fosskers committed:
(client) Quick commit before ensime testing

fosskers committed:
(client) Sidebar display extra attribute info

- Although crudely. `histogram` should be made canonical
and shown not as raw JSON

fosskers committed:
Merge pull request #1 from fosskers/extra-attributes

Extra attributes

fosskers committed:
(client) Removed debugging statements

moradology commented on client/src/main/resources/index-dev.html:
```diff
@@ -31,15 +31,15 @@
+  <script type="text/javascript" src="../../../target/scala-2.11/geotrellis-admin-client-jsdeps.js"></script>
+
+  <!-- Run the App -->
+  <script type="text/javascript" src="../../../target/scala-2.11/geotrellis-admin-client-launcher.js"></script>
```
Much nicer.

fosskers replied:
But still broken somehow, as I mentioned in Gitter.

moradology commented:
This looks good - merging in
